Source/WebKit/GPUProcess/graphics/RemoteRenderingBackend.cpp

```diff
@@ -214,18 +215,18 @@ void RemoteRenderingBackend::didCreateImageBuffer(Ref<ImageBuffer> imageBuffer)
     send(Messages::RemoteImageBufferProxy::DidCreateBackend(WTFMove(*handle)), imageBufferIdentifier);
 }
 
-void RemoteRenderingBackend::moveToSerializedBuffer(RenderingResourceIdentifier identifier)
+void RemoteRenderingBackend::moveToSerializedBuffer(RemoteSerializedImageBufferReference ref)
 {
     assertIsCurrent(workQueue());
     // Destroy the DisplayListRecorder which plays back to this image buffer.
-    m_remoteDisplayLists.take(identifier);
+    m_remoteDisplayLists.take(ref.identifier());
     // This transfers ownership of the RemoteImageBuffer contents to the transfer heap.
-    auto imageBuffer = takeImageBuffer(identifier);
+    auto imageBuffer = takeImageBuffer(ref.identifier());
     if (!imageBuffer) {
         ASSERT_IS_TESTING_IPC();
         return;
     }
-    protectedSharedResourceCache()->addSerializedImageBuffer(identifier, imageBuffer.releaseNonNull());
```

Comment:
And this is what possibly confused you: we currently don't pass use "this serialized image buffer name", because by convention we use the image buffer's old name. Since image buffer names are unique, the old name is a good name for the new serialized image buffer.